packages/components/src/product-icon/paths.jsx

```diff
@@ -0,0 +1,29 @@
+/**
+ * Internal dependencies
+ */
+import free from './images/free.svg';
```

Comment:
So right now, we'll duplicate `./images/` for each icon, and I don't see a good reason for that.

I wonder if we could reduce some of the overhead:

* Defining an `icons` array (here or even as a `const` inline with the `ProductIcon` component) with all the expected icons, no need to be an object and duplicate .
* Having one less object AND file. 
* Adding `./images/` to the beginning of the `img src`, and adding the `slug` prop afterwards, to avoid the need for specifying the paths here.
*  Simplifying `PropTypes.oneOf( Object.keys( paths ) ).isRequired,` to `PropTypes.oneOf( icons ).isRequired,`.

Reply:
> Adding `./images/` to the beginning of the `img src`, and adding the `slug` prop afterwards, to avoid the need for specifying the paths here.

I think it won't work here. We need a full path to an image in the `import` rule so that Webpack can transform this path to the actual, physical path, where the picture is located. If we remove the `./image` part, Webpack won't be able to build up the proper path.

Reply:
Good point, I think I overlooked that. Disregard 👍

Reply:
Some webpack magic like this could work:
```js
const imagePath = await import(
  /* webpackInclude: /\.svg$/ */
  /* webpackMode: "eager" */
  `./images/${ icon }`
);
```
That allows us to construct the import path dynamically and give webpack hints at compile time about what the valid values of `icon` can be.

`webpackMode: eager` instructs webpack to not create any extra chunks for the dynamic imports.

The disadvantage of this is that the `@automattic/components` library now needs webpack to work, and that the `import()` is dynamic -- it returns a promise.

The webpack-specific `require.context()` function could solve the second issue, but the first one remains.

I'm fine with the existing explicit and somewhat verbose solution, as it involves zero magic 🙂

Reply:
Yeah, exactly, I think we should try to keep it clean and with zero magic when possible 😉

Reply:
@jsnajdr isn't this using the file-loader? It just returns paths, not SVG components.

Reply:
> isn't this using the file-loader? It just returns paths, not SVG components.

Yes it is! The only thing we'd be able to tree shake are the path strings, which is not a great win. I didn't realize that.